io-engine/src/grpc/v1/stats.rs

```diff
@@ -0,0 +1,194 @@
+use crate::{
+    core::lock::ResourceLockManager,
+    grpc::{
+        rpc_submit,
+        v1::{pool::PoolService, replica::ReplicaService},
+        GrpcClientContext,
+        GrpcResult,
+        RWSerializer,
+        SvcLocker,
+    },
+    lvs::Lvs,
+};
+use futures::FutureExt;
+use io_engine_api::v1::stats::*;
+use std::fmt::Debug;
+use tonic::{Request, Response, Status};
+
+use crate::core::{BlockDeviceIoStats, CoreError, UntypedBdev};
+use ::function_name::named;
+use std::panic::AssertUnwindSafe;
+
+/// RPC service for Resource IoStats.
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct StatsService {
+    name: String,
+    client_context: tokio::sync::Mutex<Option<GrpcClientContext>>,
+    pool_svc: PoolService,
+    replica_svc: ReplicaService,
+}
+
+#[async_trait::async_trait]
+impl<F, T> SvcLocker<F, T> for StatsService
+where
+    T: Send + 'static,
+    F: core::future::Future<Output = Result<T, Status>> + Send + 'static,
+{
+    async fn svc_lock(
+        &self,
+        ctx: GrpcClientContext,
+        pool_svc: PoolService,
+        replica_svc: ReplicaService,
```

Comment:
Don't think we need a new trait, can use existing serializer trait, and just get pool and replica svc from the self

Reply:
Done. Initially thought that we dont require GrpcClientCOntext so added new trait. Then had to use it anyway for NexusService. Removed the trait now.